datadog/__init__.py

```diff
@@ -108,15 +108,32 @@ def initialize(
 
     # Statsd configuration
     # ...overrides the default `statsd` instance attributes
+    target_updated = False
```

Comment:
@valerian-roche So I'm a bit curious why we would ever want to leave sockets open if someone is calling initialize. If someone is really reinitializing the configuration, I would assume it's a low frequency event and that we can't reliably predict that leaving the socket open will be handled properly. What are your thoughts on just calling `statsd.close_socket()` before we do anything?

PS: I'm still a bit new to this code so let me know if I'm off-base with my comment

Reply:
Our code does go through it potentially multiple times for different reasons, as each parts set a part of the setup (tags, sink, ...)
The only case I could see justifying closing the socket is when host/port or socket is updated, so I isolated the case